docs/sources/flow/reference/components/otelcol.processor.resourcedetection.md

```diff
@@ -0,0 +1,939 @@
+[faas.instance][res-attr-cfg]        | Toggles the `faas.instance` resource attribute. <br> Sets `enabled` to `true` by default.        | no
+[faas.name][res-attr-cfg]        | Toggles the `faas.name` resource attribute. <br> Sets `enabled` to `true` by default.        | no
+[faas.version][res-attr-cfg]        | Toggles the `faas.version` resource attribute. <br> Sets `enabled` to `true` by default.        | no
+[gcp.cloud_run.job.execution][res-attr-cfg]        | Toggles the `gcp.cloud_run.job.execution` resource attribute. <br> Sets `enabled` to `true` by default.        | no
+[gcp.cloud_run.job.task_index][res-attr-cfg]        | Toggles the `gcp.cloud_run.job.task_index` resource attribute. <br> Sets `enabled` to `true` by default.        | no
+[gcp.gce.instance.hostname][res-attr-cfg]        | Toggles the `gcp.gce.instance.hostname` resource attribute. <br> Sets `enabled` to `false` by default.        | no
+[gcp.gce.instance.name][res-attr-cfg]        | Toggles the `gcp.gce.instance.name` resource attribute. <br> Sets `enabled` to `false` by default.        | no
+[host.id][res-attr-cfg]        | Toggles the `host.id` resource attribute. <br> Sets `enabled` to `true` by default.        | no
+[host.name][res-attr-cfg]        | Toggles the `host.name` resource attribute. <br> Sets `enabled` to `true` by default.        | no
+[host.type][res-attr-cfg]        | Toggles the `host.type` resource attribute. <br> Sets `enabled` to `true` by default.        | no
+[k8s.cluster.name][res-attr-cfg]        | Toggles the `k8s.cluster.name` resource attribute. <br> Sets `enabled` to `true` by default.        | no
+
+#### Google Compute Engine (GCE) Metadata
+
+* `cloud.provider`: `"gcp"`
+* `cloud.platform`: `"gcp_compute_engine"`
+* `cloud.account.id`: project id
+* `cloud.region`: e.g. `"us-central1"`
+* `cloud.availability_zone`: e.g. `"us-central1-c"`
+* `host.id`: instance id
+* `host.name`: instance name
+* `host.type`: machine type
+* (optional) `gcp.gce.instance.hostname`
+* (optional) `gcp.gce.instance.name`
+
+#### Google Kubernetes Engine (GKE) Metadata
+
+* `cloud.provider`: `"gcp"`
+* `cloud.platform`: `"gcp_kubernetes_engine"`
+* `cloud.account.id`: project id
+* `cloud.region`: only for regional GKE clusters; e.g. `"us-central1"`
+* `cloud.availability_zone`: only for zonal GKE clusters; e.g. `"us-central1-c"`
+* `k8s.cluster.name`
+* `host.id`: instance id
+* `host.name`: instance name; only when workload identity is disabled
+
+One known issue is when GKE workload identity is enabled, the GCE metadata endpoints won't be available, 
+thus the GKE resource detector won't be able to determine `host.name`. 
+In that case, users are encouraged to set `host.name` from either:
+- `node.name` through the downward API with the `env` detector.
+- Obtaining the Kubernetes node name from the Kubernetes API (with `k8s.io/client-go`).
```

Comment:
```suggestion
- Get the `node.name` through the downward API with the `env` detector.
- Get the Kubernetes node name from the Kubernetes API (with `k8s.io/client-go`).
```

What is a downward API.

Reply:
It is a way for a pod to get more information about the environment it is running in:
https://kubernetes.io/docs/concepts/workloads/pods/downward-api/

I'll change "downward API" to be a link to the k8s docs.